pkg/api/handlers/libpod/images.go

```diff
@@ -38,17 +38,23 @@ func ImageExists(w http.ResponseWriter, r *http.Request) {
 }
 
 func ImageTree(w http.ResponseWriter, r *http.Request) {
-	// tree is a bit of a mess ... logic is in adapter and therefore not callable from here. needs rework
-
-	// name := utils.GetName(r)
-	// _, layerInfoMap, _, err := s.Runtime.Tree(name)
-	// if err != nil {
-	//	Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrapf(err, "Failed to find image information for %q", name))
-	//	return
-	// }
-	//	it is not clear to me how to deal with this given all the processing of the image
-	// is in main.  we need to discuss how that really should be and return something useful.
-	handlers.UnsupportedHandler(w, r)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+	name := utils.GetName(r)
+
+	img, err := runtime.ImageRuntime().NewFromLocal(name)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusNotFound, errors.Wrapf(err, "Failed to find image %s", name))
+		return
+	}
+
+	whatRequires := r.Context().Value("whatrequires").(bool)
```

Comment:
2020/02/14 16:44:29 http: panic serving @: interface conversion: interface {} is nil, not bool
goroutine 229 [running]:
net/http.(*conn).serve.func1(0xc0006ca3c0)
        /usr/lib/golang/src/net/http/server.go:1767 +0x139
panic(0x18b46a0, 0xc00021b5f0)
        /usr/lib/golang/src/runtime/panic.go:679 +0x1b2
github.com/containers/libpod/pkg/api/handlers/libpod.ImageTree(0x1e48320, 0xc0002367e0, 0xc0003ac900)
        pkg/api/handlers/libpod/images.go:58 +0x405
github.com/containers/libpod/pkg/api/server.APIHandler.func1(0x1e48320, 0xc0002367e0, 0xc0003ac600)
        pkg/api/server/handler_api.go:24 +0x47e
net/http.HandlerFunc.ServeHTTP(0xc00041cec0, 0x1e48320, 0xc0002367e0, 0xc0003ac600)
        /usr/lib/golang/src/net/http/server.go:2007 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00025ecc0, 0x1e48320, 0xc0002367e0, 0xc0003ac400)
        vendor/github.com/gorilla/mux/mux.go:210 +0xe2
net/http.serverHandler.ServeHTTP(0xc0005357a0, 0x1e48320, 0xc0002367e0, 0xc0003ac400)
        /usr/lib/golang/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc0006ca3c0, 0x1e51560, 0xc000533c40)
        /usr/lib/golang/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
        /usr/lib/golang/src/net/http/server.go:2928 +0x384

Reply:
```golang
query := struct {
        Filters map[string][]string `schema:"filters"`
}{
        // override any golang type defaults
}

if err := decoder.Decode(&query, r.URL.Query()); err != nil {
        utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
                errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
        return
}
```

Is what most endpoints do.

Reply:
Ah alright thank you for the hint. Changed as suggested :+1: